src/shared/components/common/emoji-picker.tsx

```diff
@@ -0,0 +1,99 @@
+import { Component } from "inferno";
+import { i18n } from "../../i18next";
+import { getEmojiMart } from "../../utils";
+import { Icon } from "./icon";
+
+
+interface EmojiMartProps {
+    onEmojiClick?(val: any): any;
+    pickerOptions: any;
+}
+
+interface EmojiMartState {
+}
+
+export class EmojiMart extends Component<
+    EmojiMartProps,
+    EmojiMartState
+> {
+    private emptyState: EmojiMartState = {
+    };
+    constructor(props: any, context: any) {
+        super(props, context);
+        this.state = this.emptyState;
+    }
+    componentDidMount() {
+        let div: any = document.getElementById("emoji-picker");
+        if (div) {
+            div.appendChild(getEmojiMart(this.handleEmojiClick,this.props.pickerOptions));
+        }
+    }
+
+    render() {
+        return (
+            <div id="emoji-picker"></div>
+        );
+    }
+
+    handleEmojiClick = (e: any) => {
```

Comment:
Use the conventions with all the other handles, no closures. Strongly type this also.